sprintboot/src/main/java/ca/jrvs/apps/trading/dao/AccountDao.java

```diff
@@ -0,0 +1,203 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@Repository
+public class AccountDao extends CrudIntDao<Account>{
+    private JdbcTemplate jdbcTemplate;
+    private SimpleJdbcInsert insertActor;
+    private final String TABLE_NAME = "account";
+    private final String ID_COlUMN_NAME = "id";
+    private static final  String UPDATE_QUERY = "UPDATE account SET trader_id=?, amount=? " +
+            "WHERE id=?";
+    private static final  String FETCH_QUERY = "select * from account where account.id=?";
+    private static final  String FETCH_QUERY_TRADER_ID = "select * from account where trader_id=?";
+    private static final  String DELETE_QUERY = "delete from account where account.id=?";
+
+    public AccountDao(DataSource dataSource){
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.insertActor = new SimpleJdbcInsert(dataSource)
+                .withTableName(TABLE_NAME)
+                .usingGeneratedKeyColumns(this.ID_COlUMN_NAME);
+    }
+
+    public Object[] getUpdateValues(Account account){
+        Object[] objects = new Object[3];
+        objects[0] = account.getTraderId();
+        objects[1] = account.getAmount();
+        objects[2] = account.getId();
+        return objects;
+    }
+
+    @Override
+    public JdbcTemplate getJdbcTemplate() {
+        return this.jdbcTemplate;
+    }
+
+    @Override
+    public SimpleJdbcInsert getSimpleJdbcInsert() {
+        return this.insertActor;
+    }
+
+    @Override
+    public String getTableName() {
+        return this.TABLE_NAME;
+    }
+
+    @Override
+    public String getIdColumnName() {
+        return ID_COlUMN_NAME;
+    }
+
+    @Override
+    Class<Account> getEntityClass() {
+        return Account.class;
+    }
+
+    private int updateOne(Account account){
+        return this.jdbcTemplate.update(UPDATE_QUERY, this.getUpdateValues(account));
+    }
+
+    private void addOne(Account account){
+        SqlParameterSource parameterSource = new BeanPropertySqlParameterSource(account);
+        Number key = this.insertActor.executeAndReturnKey(parameterSource);
+        account.setId(key.intValue());
+    }
+
+
+
+    @Override
+    public <S extends Account> S save(S s) {
+        if(existsById((s.getId()))){
+            Integer row = this.updateOne(s);
+            if(row != 1){
+                throw new DataRetrievalFailureException("unable to update account");
+            }
+        } else {
+            this.addOne(s);
+        }
+        return s;
+    }
+
+    @Override
+    public <S extends Account> Iterable<S> saveAll(Iterable<S> iterable) {
+        List<S> accounts = new ArrayList<>();
+        iterable.forEach(accounts::add);
+        List<Object[]> batch = new ArrayList<>();
+        accounts.stream().forEach(x->{
+            if(!existsById(x.getId())){
+                throw new RuntimeException("account "+x.getId().toString() +" does not exist");
```

Comment:
try to throw a specific exception rather than Runtime

https://docs.spring.io/spring-framework/docs/current/reference/html/data-access.html#dao-exceptions